testng-core/src/main/java/org/testng/internal/DataProviderLoader.java

```diff
@@ -0,0 +1,50 @@
+package org.testng.internal;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.testng.log4testng.Logger;
+
+public class DataProviderLoader extends ClassLoader {
+  private static final int BUFFER_SIZE = 1 << 20;
+  private static final Logger log = Logger.getLogger(DataProviderLoader.class);
+
+  public Class loadClazz(String path) throws ClassNotFoundException {
+    Class clazz = findLoadedClass(path);
+    if (clazz == null) {
+      byte[] bt = loadClassData(path);
+      clazz = defineClass(path, bt, 0, bt.length);
+    }
+
+    return clazz;
+  }
+
+  private byte[] loadClassData(String className) throws ClassNotFoundException {
+    // read class
+    InputStream in = this.getClass().getClassLoader()
+        .getResourceAsStream(className.replace(".", "/") + ".class");
+    if (in == null) {
+      throw new ClassNotFoundException("Cannot load resource input stream: " + className);
+    }
+
+    byte[] classBytes = null;
```

Comment:
Cant we not just return `in.getAllBytes()` ?

Reply:
done